### DISCUSSION TITLE
Home-brew always incorrectly download x86_64 anaconda package for my M1 Mac

### DISCUSSION THREAD

**Discussion by eugeneYWang** [Homebrew/discussions#5033] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: b3751bca8c4a3c76107d5aa3b75b81db93cf4bb6
Last commit: 22 hours ago
Core tap JSON: 08 Jan 22:23 UTC
Core cask tap JSON: 08 Jan 22:23 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit westmere
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.6.2-x86_64
CLT: 15.0.0.0.1.1694021235
Xcode: 15.1
Rosetta 2: true
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  bloomrpc

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 15.1.


Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  python@2
  mongodb
  unrar

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  lua@5.3
  node@16
  openssl@1.1

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python@2
  kubernetes-cli

Warning: Some installed formulae are not readable:
  mongodb: unknown or unsupported macOS version: :mountain_lion

Warning: Your Homebrew's prefix is not /opt/homebrew.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

I was able to locate the similar issue #2434 . Tried the solution in that post but still it did not get better. 

I have right clicked my `iTerm` app and make sure it is not opened under Rosetta. 

However, `Rosetta 2` is being `true` no matter if I call ` arch -arm64 brew config | grep Rosetta ` or `brew config | grep Rosetta`.

**Comment by SMillerDev** [comment]
Is your iTerm an x86 or an ARM application? Because it seems it's the former. Try running from the normal terminal app.

**Reply by eugeneYWang** [reply]
By clicking `Get info` on the app `iTerm`, I can see it has `Kind:Application(Universal)`, and there is an option `Open using Rosetta` being off. 

I check the built-in `terminal` app under `/System/Applications/Utilities`, it has same `Kind` and `Open using Rosetta` option as well. I did not see any difference between two terminals.

Also, on both terminals, Rosetta 2 is being true no matter if I call arch -arm64 brew config | grep Rosetta or brew config | grep Rosetta.

Probably that is not relevant.

**Reply by SMillerDev** [reply]
It's definitely relevant that brew thinks it should install x86 content. I'm not sure why it does though.

I'm also pretty sure doing `arch` inside a Rosetta program won't do anything. So that would be why you can't switch back there.

**Reply by eugeneYWang** [reply]
I am sure `arch` is working under my `iTerm`. For example, with `arch`, `uname -m` shows different values

```
$  arch -arm64 uname -m
arm64
$ arch -x86_64 uname -m
x86_64
```

**Comment by eugeneYWang** [comment]
Just in case it helps: 

This my PATH

```
$ echo $PATH
/Library/Frameworks/GDAL.framework/Programs:/Users/eugene/.jabba/jdk/openjdk@1.11.0-2/Contents/Home/bin:/Users/eugene/.nvm/versions/node/v20.9.0/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/Library/Apple/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/usr/local/mysql/bin
```

**Comment by cho-m** [comment, marked as answer]
Your existing Homebrew installation is using Intel prefix, maybe from migrated Intel macOS or installation while running Rosetta:
```
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
```

Thus, your Ruby 3 is most like x86_64 so some commands will be automatically run under Rosetta:
```
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
```

---

For most users, reinstalling Homebrew using the default ARM prefix (`/opt/homebrew`) will give the best user experience. That will allow using all our pre-built bottles and avoid having to build some formulae from source.

**Reply by eugeneYWang** [reply]
Thank you for your insightful suggestions! Yes, my MacOS system was restored from a Time Machine snapshot of Intel Mac. Just like you said, reinstalling my home-brew would be helpful. 

Do you have any suggestions on reinstalling the home-brew ? I guess uninstalling home-brew and re-installing might cause a lot of dependencies or casks packages unsupported by the new brew? Or maybe it is not the best way to shift from Intel-homebrew to ARM-homebrew ?

**Reply by eugeneYWang** [reply]
Did some searchings around. Found some helpful TO-DO guidances to migrate the Intel-based home-brew to ARM-based home-brew .

I have not done these on my M1 Mac yet. Just found these to be reasonable and something I am willing to follow:

1. posts introducing Steps to backup brew cask/packages and install ARM-based brew
https://github.com/orgs/Homebrew/discussions/417#discussioncomment-601662
https://github.com/orgs/Homebrew/discussions/417#discussioncomment-1542395


2. Steps to update PATH in `.zshrc` toward ARM-based brew
https://github.com/orgs/Homebrew/discussions/417#discussioncomment-1542583

3. (optional) to sync programing packages depending on dev environments. If dev environments (like Python, Node.js) was installed and managed by brew.

 (In my case, my Node.js env is managed by `nvm`, so I don't need to worry how migration of brew will affect my Node.js env)
https://github.com/orgs/Homebrew/discussions/417#discussioncomment-1874370

4. Uninstall brew package/cask installed by intel-based brew.
https://github.com/orgs/Homebrew/discussions/417#discussioncomment-1877214

**Reply by eugeneYWang** [reply]
After going through that discussion #417 , I found that someone summarized a complete migration command set.
https://github.com/orgs/Homebrew/discussions/417#discussioncomment-2556937

(it was assuming the shell env is bash, so just use it as a reference. Not a TO-DO solution)

**Comment by eugeneYWang** [comment]
Thanks to @cho-m for guiding my way to find solutions. If you have more suggestions, I would definitely love to see them in this post
